### PR TITLE
Logging

### PR DESCRIPTION
**Pull request by jesseli2002**

### CODE REVIEW

File: avionics/common/src/log.cpp [hunk 1]
```diff
@@ -0,0 +1,137 @@
...
+ * @return Pointer to start of used buffer - this is not equal to buf; see
+ * implementation notes
+ *
+ * Getting characters out of an integer (the easy way) involves a loop of
+ * modulus & integer division, but this ends up getting the values from
+ * right to left. So, for convenience we'll just write the output from right
+ * to left, starting at the last position. This necessitates reporting what
+ * final position we ended at, hence the return value does that.
+ */
+char *itostr(char *buf, std::int32_t val) {
```
Comment (SilentSys) [inline avionics/common/src/log.cpp]: Let me know if I'm missing something, but this seems like a re-implementation of [sprintf](https://www.cplusplus.com/reference/cstdio/sprintf/). If it is indeed, I suggest we use the standard library implementation instead (widely tested, less code, will work with other types as well).

**Review by SilentSys — CHANGES_REQUESTED** [review]
Really well done. This is going to be so useful 🚀 

Just one question/suggestion before we get this merged.

File: avionics/tests/unit_Log.cpp [hunk 2]
```diff
@@ -46,24 +14,24 @@ TEST(LogOutput, Example) {
...
 
-    expect_stream << "DEBG  Filename[" << line_num + 4
-                  << "]: Debug Message 0\n";
+    // expect_stream << "DEBG  Filename[" << line_num + 4
+    //               << "]: Debug Message 0\n";
 
-    std::cout << out_stream.str(); // visualize result
-    EXPECT_EQ(out_stream.str(), expect_stream.str());
+    // // std::cout << out_stream.str(); // visualize result
+    // EXPECT_EQ(out_stream.str(), expect_stream.str());
```
Comment (SilentSys) [inline avionics/tests/unit_Log.cpp]: Is this commented out intentionally?

**Review by SilentSys — CHANGES_REQUESTED** [review]

File: avionics/tests/unit_Log.cpp (hunk 2 above)
Comment (jesseli2002) [reply avionics/tests/unit_Log.cpp]: lol nope I was debugging an issue and forgot to uncomment it

**Review by SilentSys — APPROVED** [review]

**Pull request merged**